lib/spack/spack/cmd/mirror.py

```diff
@@ -97,6 +98,19 @@ def setup_parser(subparser):
         default=spack.cmd.default_list_scope(),
         help="configuration scope to read from")
 
+    # Binary
+    upload_parser = sp.add_parser('binary', help=mirror_binary.__doc__)
+    upload_parser.add_argument('-d', '--directory', default=None,
+                               help="directory containing binaries for mirroring")
+    upload_parser.add_argument('-f', '--file', default=None,
+                               help="a file to upload to the mirror")
+    upload_parser.add_argument('-k', '--key', default=None,
+                               help="the s3 key")
+    upload_parser.add_argument('-b', '--bucket', default=None,
+                               help="name of s3 bucket")
+    upload_parser.add_argument('-p', '--profile', default=None,
```

Comment:
This looks good to me — though it should also be possible to set this in the environment (though I think that is already possible just from using boto3)